Research/lva-node-app/public/js/main.js

```diff
@@ -0,0 +1,638 @@
+//invoking methods/output related methods
+
+/** 
+* show current graph instances in dropdown. Used in output.html when invoking methods 
+* @param {HTMLButtonElement} htmlElement - button user clicks on
+*/
+function instanceMethodDropdowns(htmlElement) 
+{
+  let hoverList = htmlElement.parentElement.getElementsByTagName("ul")[0];
+  let onclickMethod = 'graphEntityModify(this, "", false, false)';
+  hoverList.innerHTML = "";
+  let checkBoxNameList = [];
+  switch(hoverList.id)
+  {
+    case 'instance-list':
+      checkBoxNameList=Object.keys(graphInstances);
+      break;
+    case 'topology-list':
+      checkBoxNameList=Object.keys(graphTopologies);
+      break;
+    case 'topology-set':
+      checkBoxNameList=Object.keys(graphTopologies);
+      onclickMethod='graphSetTopology(this)';
+      break;
+  }
+
+  checkBoxNameList.forEach((item) => 
+  {
+    hoverList.innerHTML += "<li> <button class='btn btn-outline-secondary' data-toggle='dropdown' onclick='"+onclickMethod+"' name='" + htmlElement.getAttribute('name') + "'type='checkbox'>" + item + "</button> </li>";
+  });
+}
+
+/**
+* Create the larger payload object to be sent to the server to invoke a direct method on the lvaEdge module
+* @param {string} methodNameParam - name of method to invoke
+* @param {jsonObject} methodPayload - payload for method
+* @returns {jsonObject} full payload for request
+*/
+function createFullPayload(methodNameParam, methodPayload)
+{
+  let fullPayload = 
+  {
+    methodName: methodNameParam,
+    responseTimeoutInSeconds: 200,
+    Payload: methodPayload  
+  }
+  return fullPayload;
+}
+
+/**
+* see sample graph topologies here: https://github.com/Azure/live-video-analytics/tree/master/MediaGraph/topologies
+* @param {string | HTMLButtonElement} htmlElement - either a string graph name or the HTML button containing the name of the graph
+* @param {boolean} - if name is passed = true
+* @returns {void}
+*/
+function graphSetTopology(htmlElement, nameIsPassed=false)
+{
+  let topologyName = nameIsPassed ? htmlElement : htmlElement.innerText;
+
+  let topology = graphTopologies[topologyName];
+  if(topology == undefined)
+  {
+    alert("Topology "+topologyName+" not found.");
+    return;
+  }
+  invokeLVAMethod(createFullPayload("GraphTopologySet", topology));
+}
+
+/**
+* invoke a list method on lvaEdge module (either list topology or list instances. payload is the same)
+* @param {HTMLButtonElement} htmlElement - either a string of method name, or the HTML button clicked to invoke method
+* @param {boolean} - if name is passed = true
+* @returns {Promise<any>}
+*/
+function graphEntityList(htmlElement, nameIsPassed=false) 
+{
+  return new Promise((resolve, reject) =>
+  {
+    var methodName = nameIsPassed ? htmlElement : htmlElement.getAttribute('name');
+    let payload = 
+    {
+      "@apiVersion": "1.0"
+    }
+    invokeLVAMethod(createFullPayload(methodName, payload)).then((response) =>
```

Comment:
above we don't resolve... why are these two very similar methods different?

Reply:
fixed all instances of .then + .catch to use the same semantic